CMakeLists.txt

```diff
@@ -0,0 +1,98 @@
+option (IMGUI_IMPL_DX9 "Build the DirectX 9 implementation (only if supported)" ${WIN32})
+option (IMGUI_IMPL_DX10 "Build the DirectX 10 implementation (only if supported)" ${WIN32})
+option (IMGUI_IMPL_DX11 "Build the DirectX 11 implementation (only if supported)" ${WIN32})
+option (IMGUI_IMPL_DX12 "Build the DirectX 12 implementation (only if supported)" ${WIN32})
+
+add_library (imgui
+    imgui.cpp
+    imgui_draw.cpp
+    imgui_widgets.cpp
+    $<$<BOOL:IMGUI_DEMO>:imgui_demo.cpp>
+)
+
+add_library(imgui::imgui ALIAS imgui)
+
+target_include_directories (imgui PUBLIC "${CMAKE_CURRENT_SOURCE_DIR}")
+
+target_compile_features(imgui PRIVATE cxx_std_98)
+
+if (IMGUI_EXAMPLES)
+    # Sets up polyfill libraries for Windows examples (e.g. GLFW)
+    include (examples/libs/glfw.cmake)
+endif ()
+
+if (NOT IMGUI_DEMO)
+    target_compile_definitions (imgui PUBLIC -DIMGUI_DISABLE_DEMO_WINDOWS=1)
+endif ()
+
+if (IMGUI_IMPL_SDL)
+    include (examples/imgui_impl_sdl.cmake)
+endif ()
+if (IMGUI_IMPL_METAL)
+    include (examples/imgui_impl_metal.cmake)
+endif ()
+if (IMGUI_IMPL_OSX)
+    include (examples/imgui_impl_osx.cmake)
+endif ()
+if (IMGUI_IMPL_WIN32)
+    include (examples/imgui_impl_win32.cmake)
+endif ()
+if (IMGUI_IMPL_GLFW)
+    include (examples/imgui_impl_glfw.cmake)
+endif ()
+if (IMGUI_IMPL_OPENGL OR IMGUI_IMPL_OPENGL2)
+    include (examples/imgui_impl_opengl.cmake)
+endif ()
+if (IMGUI_IMPL_GLUT)
+    include (examples/imgui_impl_glut.cmake)
+endif ()
+if (IMGUI_IMPL_DX9)
+    include (examples/imgui_impl_dx9.cmake)
+endif ()
+if (IMGUI_IMPL_DX10)
+    include (examples/imgui_impl_dx10.cmake)
+endif ()
+if (IMGUI_IMPL_DX11)
+    include (examples/imgui_impl_dx11.cmake)
+endif ()
+if (IMGUI_IMPL_DX12)
+    include (examples/imgui_impl_dx12.cmake)
+endif ()
+
+if (IMGUI_EXAMPLES)
+    set (IMGUI_EXAMPLE_OUTPUT_DIR "${CMAKE_CURRENT_BINARY_DIR}")
+
+    #add_subdirectory (examples/example_allegro5)
+    add_subdirectory (examples/example_apple_metal)
+    add_subdirectory (examples/example_apple_opengl2)
+    #add_subdirectory (examples/example_emscripten)
+    add_subdirectory (examples/example_glfw_metal)
+    add_subdirectory (examples/example_glfw_opengl2)
+    add_subdirectory (examples/example_glfw_opengl3)
+    #add_subdirectory (examples/example_glfw_vulkan)
+    add_subdirectory (examples/example_glut_opengl2)
+    add_subdirectory (examples/example_null)
+    #add_subdirectory (examples/example_sdl_directx11)
+    add_subdirectory (examples/example_sdl_metal)
+    add_subdirectory (examples/example_sdl_opengl2)
+    add_subdirectory (examples/example_sdl_opengl3)
+    #add_subdirectory (examples/example_sdl_vulkan)
+    add_subdirectory (examples/example_win32_directx10)
+    add_subdirectory (examples/example_win32_directx11)
+    add_subdirectory (examples/example_win32_directx12)
+    add_subdirectory (examples/example_win32_directx9)
```

Comment:
Maybe instead of many add_subdirectory calls here, we can move them to "examples/CMakeLists.txt"?

Reply:
I recall there being a reason I didn't do that initially, but otherwise I agree.